libnetdata/socket/socket.c

```diff
@@ -846,18 +927,28 @@ int connect_to_one_of(const char *destination, int default_port, struct timeval
         // is there anything?
         if(!*s || s == e) break;
 
-        char buf[e - s + 1];
-        strncpyz(buf, s, e - s);
-        if(reconnects_counter) *reconnects_counter += 1;
-        sock = connect_to_this(buf, default_port, timeout);
-        if(sock != -1) {
-            if(connected_to && connected_to_size) {
-                strncpy(connected_to, buf, connected_to_size);
-                connected_to[connected_to_size - 1] = '\0';
+        if (i < 32)
+            connect = !(*do_not_connect & (1U<<i));
+        else
+            connect = 1;
+
+        if(connect) {
+            char buf[e - s + 1];
+            strncpyz(buf, s, e - s);
+            if(reconnects_counter) *reconnects_counter += 1;
+            sock = connect_to_this(buf, s, default_port, timeout);
+            if(sock == -2)
+                *do_not_connect |=  (1U<<i);
```

Comment:
When I specify a wrong host (as detected by `test_tcp_hostname`) it returns -1 and the `do_not_connect` is not set
example would be to try to stream to `192.168.0._1` 

```
2020-02-13 13:49:59: netdata ERROR : STREAM_SENDER[debian1] : One of the values specified is not valid host = '192.168.122._1', service = '19999', interface = '', protocol = 6
2020-02-13 13:49:59: netdata ERROR : STREAM_SENDER[debian1] : STREAM debian1 [send to 192.168.122._1:19999]: failed to connect
```

Reply:
Probably the name is not good, but this is a variable used to identify the address that is invalid and we won't connect again. This is used as a flag, where each one of the 32 positions identify one of the 32 possible address in the list that tried to reconnect. Every time that the address is invalid(return -2 ) I enable the bit and I do not try to reconnect again. Should we rename the variable?